### DISCUSSION TITLE
DataGrid cell background and foreground colour

### DISCUSSION THREAD

**Discussion by abogadi** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3324] [Q&A]
I am trying to change the background of individual cells in a DataGrid column based on their content. 
Th below simplified code does not seem to play together with Material Design in Xaml (It does work without Material Design in Xaml)

Any ideas on how does one do with Material Design in Xaml applied?

```
textColumn.ElementStyle = new Style(targetType: typeof(TextBlock));

textColumn.ElementStyle.Setters.Add(item: new Setter(property: TextBlock.ForegroundProperty,
                                                     value: Brushes.White ));
textColumn.ElementStyle.Setters.Add(item: new Setter(property: TextBlock.BackgroundProperty,
                                                     value: Brushes.Red  ));
```

**Comment by nicolaihenriksen** [comment]
@abogadi I think it very much depends on when the code you provided is being executed. Material Design in XAML will attempt to push its own styles onto some of the standard column types, unless the default has been manipulated beforehand (which is probably what you want).

Can you share a bit of the XAML you have and also some context around the code in the description above? Thanks.
